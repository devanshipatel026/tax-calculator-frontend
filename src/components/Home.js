import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import { getCalculatedTax } from "../services/tax";

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      income: "",
      fields: {},
      errors: {}
    };
  }

  submitForm(e) {
    e.preventDefault();
    getCalculatedTax(this.state.income).then((response) =>{ 
    if(response.message === 'Success'){
      this.props.history.push({
        pathname: "/CalculateTax",
        state: { income: this.state.income, bracket: response.bracket, totalTax: response.totalTax },
      });
    }else(
      console.log("Error retrieving Tax bracket!")
    )
  });
  }

  render() {
    return (
      <div>
        <form className="Home" onSubmit={this.submitForm.bind(this)}>
          <div className="form-group row">
            <label className="Home-text col-sm-2 col-form-label mt-2">
              Enter Gross Income:
            </label>
            <div className="col-sm-6 form-group row">
              <input
                required
                className="form-control mt-2"
                id="income"
                type="number"
                placeholder="Enter Gross Income (in CAD)"
                value={this.state.income}
                onChange={(e) => this.setState({ income: e.target.value })}
              />
              <button type="submit" className="btn btn-info mt-4">
                Calculate
              </button>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

export default withRouter(Home);
