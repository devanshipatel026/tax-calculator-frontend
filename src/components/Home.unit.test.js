import React from 'react';
import { render } from '@testing-library/react';
import  Home  from './Home';
import renderer from 'react-test-renderer';
import { shallow,  configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

//snapshot rendering branch
describe('Home', () => {
    test('snapshot renders', () => {
      const component = renderer.create(<Home.WrappedComponent />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

describe('Home', () => {
  it('initial state', () => {
    render(<Home.WrappedComponent/>)
    // it renders empty income fields
    const wrapper = shallow(<Home.WrappedComponent/>);
    const text = wrapper.find('input').text();
    expect(text).toEqual("");
  });


});