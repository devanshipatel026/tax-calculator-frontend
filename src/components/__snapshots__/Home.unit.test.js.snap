// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot renders 1`] = `
<div>
  <form
    className="Home"
    onSubmit={[Function]}
  >
    <div
      className="form-group row"
    >
      <label
        className="Home-text col-sm-2 col-form-label mt-2"
      >
        Enter Gross Income:
      </label>
      <div
        className="col-sm-6 form-group row"
      >
        <input
          className="form-control mt-2"
          id="income"
          onChange={[Function]}
          placeholder="Enter Gross Income (in CAD)"
          required={true}
          type="number"
          value=""
        />
        <button
          className="btn btn-info mt-4"
          type="submit"
        >
          Calculate
        </button>
      </div>
    </div>
  </form>
</div>
`;
